lua/entities/gmod_wire_turret.lua

```diff
@@ -1,182 +1,219 @@
+end
+
+function ENT:SetDelay( delay )
+	local check = game.SinglePlayer() -- clamp delay if it's not single player
+	local limit = check and 0.01 or 0.05
+	self.delay = math.Clamp( delay, limit, 1 )
+end
+
+function ENT:SetNumBullets( numbullets )
+	local check = game.SinglePlayer() -- clamp num bullets if it's not single player
+	local limit = math.floor( math.max( 1, numbullets ) )
+	self.numbullets = check and limit or math.Clamp( limit, 1, 10 )
+end
+
+function ENT:SetTracer( tracer )
+	local tracer = string.Trim(tracer)
+	self.tracer = ValidTracers[tracer] and tracer or ""
+end
+
+function ENT:SetSpread( spread )
+	self.spread = math.Clamp( spread, 0, 1 )
+	self.spreadvector.x = self.spread
+	self.spreadvector.y = self.spread
+end
+
+function ENT:SetDamage( damage )
+	self.damage = math.Clamp( damage, 0, 100 )
+end
+
+function ENT:SetBlastDamage( blastdamage )
+	self.blastdamage = math.Clamp( blastdamage, 0, GetConVar("wire_turret_blastdamage_max") )
+end
+
+function ENT:SetBlastRadius( blastradius )
+	self.blastradius = math.Clamp( blastradius, 0, GetConVar("wire_turret_blastradius_max") )
+end
+
+function ENT:SetForce( force )
+	self.force = math.Clamp( force, 0, 500 )
+end
+
+function ENT:SetTraceNum( tracernum )
+	self.tracernum = math.floor( math.max( tracernum or 1 ) )
+end
+
+function ENT:TriggerInput( iname, value )
+	if (iname == "Fire") then
+		self.Firing = value > 0
+	elseif (iname == "Force") then
+		self:SetForce( value )
+	elseif (iname == "Damage") then
+		self:SetDamage( value )
+	elseif (iname == "NumBullets") then
+		self:SetNumBullets( value )
+	elseif (iname == "Spread") then
+		self:SetSpread( value )
+	elseif (iname == "Delay") then
+		self:SetDelay( value )	
+	elseif (iname == "Blast Damage") then
+		self:SetBlastDamage( value )		
+	elseif (iname == "Blast Radius") then
+		self:SetBlastRadius( value )
+	elseif (iname == "Sound") then
+		self:SetSound( value )
+	elseif (iname == "Tracer") then
+		self:SetTracer( value )
+	end
+end
+
+function ENT:Setup(delay, damage, force, sound, numbullets, spread, blastdamage, blastradius, tracer, tracernum)
+	self:SetForce(force)
+	self:SetDelay(delay)
+	self:SetSound(sound)
+	self:SetDamage(damage)
+	self:SetSpread(spread)
+	self:SetBlastDamage(blastdamage)
+	self:SetBlastRadius(blastradius)
+	self:SetTracer(tracer)
+	self:SetTraceNum(tracernum)
+	self:SetNumBullets(numbullets)
+end
+
+-- Legacy Support
+function ENT:Setup(delay, damage, force, sound, numbullets, spread, tracer, tracernum)
+	self:SetForce(force)
+	self:SetDelay(delay)
+	self:SetSound(sound)
+	self:SetDamage(damage)
+	self:SetSpread(spread)
+	self:SetTracer(tracer)
+	self:SetTraceNum(tracernum)
+	self:SetNumBullets(numbullets)
+end
+
+duplicator.RegisterEntityClass( "gmod_wire_turret", WireLib.MakeWireEnt, "Data", "delay", "damage", "force", "sound", "numbullets", "spread", "blastdamage", "blastradius", "tracer", "tracernum" )
+-- Legacy Support

```

Comment:
Once again not how this works, pretty sure this would just overwrite the other one.

Reply:
If you want to fix this stuff and have proper legacy support you need to put the blast damage / radius inputs after the trace ones